mitodl/open-discussions
# Details page + AddToListDialog for stafflists

#### Pre-Flight checklist


#### What are the relevant tickets?
#3959 

#### What's this PR do?
This PR adds a details page and the "AddToListDialog" for stafflists.

#### How should this be manually tested?
1. View `http://od.odl.local:8063/infinite/stafflists`. If you don't have any, create some new stafflists (and paths)
2. Add items to stafflists. This can be done on any page with LearningResourceCards. The `/infinite` and `/infinite/search` are good options.
    - On a card, click bookmark icon to add resource to **userlist**.  ("Add to userlist dialog")
    - On a card, click list icon icon to add resource to **stafflist**. ("Add to stafflist dialog")
3. Add several resources to your userlists. Verify the dialogs above contain only userslists or only stafflists. While you're at it, try the "Create new list" button in the dialogs.
4. View `http://od.odl.local:8063/infinite/stafflists`; Click any stafflist to view the details page. The resources you added should be visible on the list.
5. For a staff path ("MIT Learning List") try re-ordering the items.
6. Try removing items from a stafflist using the "Add/remove to stafflist dialog".

#### Notes / Issues
- 403 handling for private stafflists will be part of #3864 
- ⚠️ You might notice that while viewing the stafflist details, if you click bookmark icon to add to a userlist, the bookmark icon is not filled immediately. This will be handled as part of #3956 
- At least for now, the "stafflist icon" on resource cards is a single icon; there is no filled vs unfilled like there is for bookmark.

#### Screenshots (if appropriate)

<img width="400" alt="Screenshot 2023-05-05 at 5 48 01 AM" src="https://user-images.githubusercontent.com/9010790/236429539-9b7c34c6-b1dd-41b9-a420-913003cae8d4.png">
<img width="400" alt="Screenshot 2023-05-05 at 5 48 16 AM" src="https://user-images.githubusercontent.com/9010790/236429548-d090d120-3448-42e7-800a-2cc2d103c8e8.png">
<img width="400" alt="Screenshot 2023-05-05 at 5 50 26 AM" src="https://user-images.githubusercontent.com/9010790/236429556-c75c184a-ce79-482f-b4da-ce687c2d1c14.png">
